resources/iam-user-policy.go

```diff
@@ -0,0 +1,64 @@
+package resources
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/iam"
+)
+
+type IAMUserPolicy struct {
+	svc        *iam.IAM
+	userName   string
+	policyName string
+}
+
+func init() {
+	register("IAMUserPolicy", ListIAMUserPolicies)
+}
+
+func ListIAMUserPolicies(sess *session.Session) ([]Resource, error) {
+	svc := iam.New(sess)
+
+	users, err := svc.ListUsers(nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resources := make([]Resource, 0)
+	for _, user := range users.Users {
+		policies, err := svc.ListUserPolicies(&iam.ListUserPoliciesInput{
+			UserName: user.UserName,
+		})
+		if err != nil {
+			return nil, err
+		}
+
+		for _, policyName := range policies.PolicyNames {
+			resources = append(resources, &IAMUserPolicy{
+				svc:        svc,
+				policyName: *policyName,
+				userName:   *user.UserName,
+			})
+		}
+	}
+
+	return resources, nil
+}
+
+func (e *IAMUserPolicy) Remove() error {
```

Comment:
Why `e` though?

Reply:
its named like this in other places as well. for example: 
https://github.com/rebuy-de/aws-nuke/blob/2d9b2f8fe3937ae7972a9ccc41159470525778f7/resources/iam-users.go#L36
https://github.com/rebuy-de/aws-nuke/blob/2d9b2f8fe3937ae7972a9ccc41159470525778f7/resources/iam-user-policy-attachments.go#L52
i'm also not a big fan of those ambiguous one letter variable names but i felt like following the style of the other resources out-weights that urge in that case.
if we adapt to having proper variable naming here, then i'd vote for doing it in all the places at once in a separate pull request.

Reply:
FYI https://github.com/golang/go/wiki/CodeReviewComments#receiver-names